pulp-glue/pulp_glue/common/openapi.py

```diff
@@ -252,24 +260,26 @@ def _parse_api(self, data: bytes) -> None:
 
     def _download_api(self) -> bytes:
         try:
-            response: requests.Response = self._session.get(urljoin(self.base_url, self.doc_path))
+            response: requests.Response = self._session.get(urljoin(self._base_url, self._doc_path))
         except requests.RequestException as e:
             raise OpenAPIError(str(e))
         response.raise_for_status()
-        if "Correlation-ID" in response.headers:
-            self._set_correlation_id(response.headers["Correlation-ID"])
+        if "Correlation-Id" in response.headers:
```

Comment:
Having recently been burned by this - is there a good way to make this a case-insensitive comparison?

Reply:
I'll look into CIMultiDict.

Reply:
Instead of reworking it here, i decided to add some more ready commits that already included this improvement.